### ISSUE TITLE
backup Button On PyLeoRef.leo Does Not Work

### ISSUE THREAD

**Issue by tbpassin** [leo-editor/leo-editor#3165]
This button failed with the error message `base_dir not found: None`.  The problem is that the button calls `c.backup_helper(sub_dir='leoPy')` but although the method signature seems make the argument _base_dir_ optional, it actually checks for whether or not the arg is _None_.  If so, it doesn't do anything except to emit the message.

**Comment by edreamleo** [comment]
@tbpassin I'll attempt a fix today.

**Comment by edreamleo** [comment]
@tbpassin The logic in `c.backup_helper` isn't the best, but imo the helper does all that it can. The method should *not* just pick a place at random to write back up files!

`c.backup_helper` does more than just check the `base_dir` kwarg. The docstring clearly states that `c.backup_helper` attempts to use `os.environ[env_key]` when `base_dir` is None. Furthermore, `env_key` defaults to the string `"LEO_BACKUP"`.

The "workaround" for you will be to set `os.environ["LEO_BACKUP"]` to an existing directory.

HTH.

**Comment by edreamleo** [comment]
@tbpassin Rev 91ba238 in devel adds the following docstring for `@button backup` in LeoPyRef.leo`:
```python
"""
Back up this .leo file.

os.environ['LEO_BACKUP'] must be the path to an existing (writable) directory.
"""
```
You were quite right to complain about the lack of the docstring :-)

The diffs are too large, which is an ongoing problem.

**Comment by tbpassin** [comment]
I forgot to mark this as a "Buglet" instead of a "Bug".  Yes, after I went to the command def, I saw that an environmental value can be set.  But with a subdir name of "leoPy", I assumed that the main purpose was to create a copy of LeoPyRef named LeoPy, which (to me) would logically go where ever  the "LeoPy" menu item in the _File/Open Specific Leo File_ would point to.

I'd like to see, at a minimum, an error message that explained that the _LEO_BACKUP_  variable should be set.  Maybe I'll get to that at some point, not a high priority.
